### PR TITLE
Allow selection of ensemble in top nav bar

### PR DESCRIPTION
**Pull request by sigurdp**
Added ensemble selection in nav bar, for now just immediate selection of a single ensemble.
Selected ensemble is communicated through `WorkbenchServices` on the `navigator.ensembles` topic.
Communication is through an array of ensembles, but currently only an empty array or an array with a single item will be communicated.
Adapted `SimulationTimeSeries` to use the ensemble selection from the nav bar.

### CODE REVIEW

File: frontend/src/modules/SimulationTimeSeries/settings.tsx [hunk 1]
```diff
@@ -4,81 +4,53 @@ import { Frequency, VectorDescription } from "@api";
...
-    function makeFrequencyListItems(): { label: string; value: string }[] {
-        const itemArr: { label: string; value: string }[] = [
-            { value: Frequency.DAILY, label: "Daily" },
-            { value: Frequency.MONTHLY, label: "Monthly" },
-            { value: Frequency.QUARTERLY, label: "Quarterly" },
-            { value: Frequency.YEARLY, label: "Yearly" },
-            { value: "RAW", label: "None (raw)" },
-        ];
-        return itemArr;
+        const newName = selectedVecNames.length > 0 ? selectedVecNames[0] : "";
```
Comment (rubenthoms) [inline frontend/src/modules/SimulationTimeSeries/settings.tsx:52]: Could be `selectedVecNames[0] ?? ""`

**Review by rubenthoms — CHANGES_REQUESTED** [review]
Some suggestions, one requested change. Otherwise, LGTM :+1:

File: frontend/src/modules/SimulationTimeSeries/settings.tsx [hunk 2]
```diff
@@ -4,81 +4,53 @@ import { Frequency, VectorDescription } from "@api";
...
 //-----------------------------------------------------------------------------------------------------------
 export function settings({ moduleContext, workbenchServices }: ModuleFCProps<State>) {
-    console.log("render Settings");
+    console.log("render SimulationTimeSeries settings");
 
-    const caseUuid = useSubscribedValue("navigator.caseId", workbenchServices);
-    const [ensembleName, setEnsembleName] = moduleContext.useStoreState("ensembleName");
-    const [vectorName, setVectorName] = moduleContext.useStoreState("vectorName");
+    const ensembles = useSubscribedValue("navigator.ensembles", workbenchServices);
+    const [userSelectedVectorName, setUserSelectedVectorName] = React.useState("");
```
[the remark this answers is not in the document]
Comment (sigurdp) [reply frontend/src/modules/SimulationTimeSeries/settings.tsx]: Fixed

File: frontend/src/modules/SimulationTimeSeries/settings.tsx [hunk 3]
```diff
@@ -4,81 +4,53 @@ import { Frequency, VectorDescription } from "@api";
...
-
-    const vectorsQuery = useQuery({
-        queryKey: ["getVectorNamesAndDescriptions", caseUuid, ensembleName],
-        queryFn: () => apiService.timeseries.getVectorNamesAndDescriptions(caseUuid ?? "", ensembleName ?? ""),
-        enabled: caseUuid && ensembleName ? true : false,
-        onSuccess: function selectDefaultVector(vectorArr) {
-            console.log("selectDefaultVector()");
-            if (vectorArr.length > 0) {
-                setVectorName(vectorArr[vectorArr.length - 1].name);
+    const firstEnsemble = ensembles && ensembles.length > 0 ? ensembles[0] : null;
```
[the remark this answers is not in the document]
Comment (sigurdp) [reply frontend/src/modules/SimulationTimeSeries/settings.tsx]: Fixed

File: frontend/src/modules/SimulationTimeSeries/settings.tsx (hunk 1 above)
Comment (sigurdp) [reply frontend/src/modules/SimulationTimeSeries/settings.tsx]: Updated this

File: frontend/src/modules/SimulationTimeSeries/settings.tsx [hunk 4]
```diff
@@ -4,81 +4,53 @@ import { Frequency, VectorDescription } from "@api";
...
-        if (vectorsQuery.isSuccess && vectorsQuery.data.length > 0) {
-            for (const vec of vectorsQuery.data) {
-                itemArr.push({ value: vec.name, label: vec.descriptive_name });
-            }
-        } else {
-            // itemArr.push({ value: "", label: `${vectorsQuery.status.toString()}...`, disabled: true });
-        }
-        return itemArr;
-    }
+        [firstEnsemble?.caseUuid, firstEnsemble?.ensembleName, renderVectorName]
```
[the remark this answers is not in the document]
Comment (sigurdp) [reply frontend/src/modules/SimulationTimeSeries/settings.tsx]: Agree

**Pull request merged**